### DISCUSSION TITLE
graphql requests are taken as async supported eventhough the processing is synchronous with 11.0.0 version

### DISCUSSION THREAD

**Discussion by mutyasaisrikar** [graphql-java-kickstart/graphql-spring-boot#539] [Q&A]
We were using `graphql-spring-boot-starter 6.0.0` and zalando logbook to log http requests and responses at the end of the request processing.

But when upgraded to `graphql-spring-boot-starter 11.0.0`, the logbook filter post process is not getting executed because GraphQL is trying to start the request in an async way.

When I tried to debug the flow, I can see that HttpRequestInvokerImpl.execute method is getting executed now which is doing this `request.startAsync(request, response);`.

This was not happening with 6.0.0 version.

How can I disable this from happening so that my flow happens synchronously and my logbook will be executed.

**Comment by mutyasaisrikar** [comment]
@oliemansm  `graphql.servlet.asyncModeEnabled` is removed from `GraphQLServletProperties.java` after 8.0.0 version. How to disable this with 11.0.0?

**Reply by oliemansm** [reply]
@mutyasaisrikar The way the async mode works has been refactored (as you've noticed). This resulted in the async mode being used when the servlet has been configured to support async mode. That property was removed in the process, which resulted in it being always async.

I've registered an enhancement #540 to reinstate that property, although the default value now has been inverted. So it'll be enabled by default, but it'll be easy to disable by setting `graphql.servlet.async-mode-enabled` to `false`. It'll be part of 11.1.0.

**Reply by mutyasaisrikar** [reply]
@oliemansm, is there any update on the release of this version?

**Reply by fmartins** [reply]
Hello; There are alternatives to continue using the GraphQL async mode with the Spring Request Context implementation based on Thread Locals?

I'm having the exception below while trying to set properties on Request context in a normal Servlet Filter.
> java.lang.IllegalStateException: No thread-bound request found: Are you referring to request attributes outside of an actual web request, or processing a request outside of the originally receiving thread? If you are actually operating within a web request and still receive this message, your code is probably running outside of DispatcherServlet: In this case, use RequestContextListener or RequestContextFilter to expose the current request.

Reference: https://ajurasz.github.io/2019-12-28-accessing-request-data-from-outside-of-calling-thread/
